src/components/ColumnItem.tsx

```diff
@@ -0,0 +1,113 @@
+import { useState } from 'react'
+import { useFirestore } from 'react-redux-firebase'
+import { useSelector } from 'react-redux'
+import {
+  Paper,
+  IconButton,
+  TextField,
+  Typography,
+  Divider,
+} from '@mui/material'
+import CheckIcon from '@mui/icons-material/Check'
+import DeleteIcon from '@mui/icons-material/Delete'
+import EditIcon from '@mui/icons-material/Edit'
+import { CardList } from './CardList'
+import { CardForm } from './CardForm'
+import { Card, WithID, RootState } from '../store/store'
+
+export const ColumnItem = (props: {
+  cards: WithID<Card>[]
+  title: string
+  id: string
+  key: string
+}) => {
+  const firestore = useFirestore()
+  const { uid } = useSelector((state: RootState) => state.firebase.auth)
+
+  const columnCards = props.cards.filter((card) =>
+    card ? card.columnID === props.id : false
+  )
+
+  const [editMode, setEditMode] = useState(false)
+  const [presetColumnTitle, setPresetColumnTitle] = useState(props.title)
+
+  const handleChange = ({
+    currentTarget: { name, value },
+  }: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    if (name === 'editColumn') {
+      setPresetColumnTitle(value)
+    }
+  }
+
+  const editColumn = (columnTitle: string) => {
+    firestore
+      .collection('users')
+      .doc(uid)
+      .collection('columns')
+      .doc(props.id)
+      .update({ title: columnTitle })
+
+    setEditMode(false)
+  }
+
+  const deleteColumn = () => {
+    firestore
+      .collection('users')
+      .doc(uid)
+      .collection('columns')
+      .doc(props.id)
+      .delete()
+  }
```

Comment:
I would suggest moving these to a separate file. It is very possible that you may need to call the same functions in different places. Also, it will simplify the component. You can move it temporarily to, for example, `src/firebase/services.ts`. You can also leave it and as we will probably improve whole app structure in the future. Up to you.